### PR TITLE
Add Solr ops for vertical scaling and volume expansion

### PR DESCRIPTION
**Pull request by pritamdas99**

### CODE REVIEW

**Review by raihankhan — APPROVED** [review]
LGTM 👍🏻

File: apis/kubedb/constants.go [hunk 1]
```diff
@@ -923,6 +927,15 @@ const (
 	SolrConfAllowPathsValue        = ""
 	SolrConfSolrCloudKey           = "solrcloud"
 	SolrConfShardHandlerFactoryKey = "shardHandlerFactory"
+
+	ProxyDeploymentName = "s3proxy"
+	ProxyServiceName    = "proxy-svc"
+	ProxySecretName     = "proxy-env"
+	ProxyImage          = "andrewgaul/s3proxy"
```
Comment (ArnobKumarSaha) [inline apis/kubedb/constants.go:934]: This repo has not been cloned ?

File: apis/kubedb/v1alpha2/solr_helpers.go [hunk 2]
```diff
@@ -263,6 +266,15 @@ func (s *Solr) SetDefaults(slVersion *catalog.SolrVersion) {
 		}
 	}
 
+	var slVersion catalog.SolrVersion
+	err := DefaultClient.Get(context.TODO(), types.NamespacedName{
+		Name: s.Spec.Version,
+	}, &slVersion)
+	if err != nil {
+		klog.Errorf("can't get the rabbitmq version object %s for %s \n", err.Error(), s.Spec.Version)
```
Comment (ArnobKumarSaha) [inline apis/kubedb/v1alpha2/solr_helpers.go:274]: solr version

File: apis/ops/v1alpha1/solr_ops_types.go [hunk 3]
```diff
@@ -58,15 +59,72 @@ type SolrOpsRequestSpec struct {
...
+	ConfigSecret *core.LocalObjectReference `json:"configSecret,omitempty"`
+	// ApplyConfig is an optional field to provide rabbitmq configuration.
+	// Provided configuration will be applied to config files stored in ConfigSecret.
+	// If the ConfigSecret is missing, the operator will create a new k8s secret by the
+	// following naming convention: {db-name}-user-config .
+	// Expected input format:
+	//	applyConfig:
+	//		rabbitmq.conf: |
+	//			key=value
+	// +optional
```
Comment (ArnobKumarSaha) [inline apis/ops/v1alpha1/solr_ops_types.go:120]: `solr.xml`

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: apis/kubedb/constants.go (hunk 1 above)
Comment (pritamdas99) [reply apis/kubedb/constants.go:934]: no

**Review by ArnobKumarSaha — APPROVED** [review]

**Pull request merged**